kotlin/src/main/kotlin/com/linecorp/armeria/internal/common/kotlin/FlowCollectingPublisher.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License
+ */
+
+package com.linecorp.armeria.internal.common.kotlin
+
+import com.linecorp.armeria.common.stream.DefaultStreamMessage
+import kotlinx.coroutines.AbstractCoroutine
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.handleCoroutineException
+import org.reactivestreams.Publisher
+import org.reactivestreams.Subscriber
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * [Publisher] implementation which emits values collected from [Flow].
+ * Reactive streams back-pressure works on its backing [flow] too.
+ */
+@OptIn(InternalCoroutinesApi::class)
+internal class FlowCollectingPublisher<T : Any>(
+    private val flow: Flow<T>,
+    context: CoroutineContext = EmptyCoroutineContext
+) : Publisher<T>, AbstractCoroutine<Unit>(context, initParentJob = false, active = true) {
+    private val delegate = DefaultStreamMessage<T>()
+
+    override fun subscribe(s: Subscriber<in T>) {
+        this.start(CoroutineStart.DEFAULT, this) {
+            flow.collect {
+                delegate.write(it)
+                delegate.whenConsumed().await()
+            }
+            delegate.close()
+        }
+        delegate.whenComplete().handle { _, _ -> cancel() }
+        delegate.subscribe(s)
```

Comment:
How about specifying `ServiceRequestContext.eventLoop()` when subscribing to `delegate`?
```suggestion
        delegate.subscribe(s, ctx.eventLoop())
```

Reply:
Made the constructor receive `ServiceRequestContext` as a parameter so that I can do this!